arroyo-worker/src/connectors/sse.rs

```diff
@@ -188,15 +197,7 @@ where
                                             let iter = self.deserializer.deserialize_slice(&event.data.as_bytes()).await;
 
                                             for v in iter {
-                                                match v {
-                                                    Ok(value) => {
-                                                        ctx.collector.collect(Record {
-                                                            timestamp: SystemTime::now(),
-                                                            key: None,
-                                                            value,
-                                                        }).await;
-                                                    }
-                                                    Err(e) => {
+                                                if let Err(e) = collect_source_record(ctx, SystemTime::now(), v, &self.bad_data, &mut self.rate_limiter).await {
```

Comment:
Same here — there's existing error handling code that is now duplicative with what collect_source_record is doing

Reply:
I've removed this